### ISSUE TITLE
Finalize GNNs for the scientific paper

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/3D-Vac#151]
We use here newly generated data (see issue #149), with no pssm (the files we have are wrong) and with the newest software version (15/05/2023), to finalize GNNs training for the research paper, always with the 100k pMHCI data (see issue #113).
- Experiments are saved in `/data/pMHCI/trained_models/deeprank2/experiments/`
- HDF5 files are in `/data/pMHCI/features_output_folder/deeprank2/residue/230530`
   - graphs only, 100069
    - Using `230515` in experiment `exp_100k_std_bs64_net1bn_0_230526` (46 epochs run), the total time was ~13h (1 GPU, 16 workers). Using a dataset without grids it took ~4h with similar parameters. So for time convenience, we will use a dataset containing graphs only for GNNs, and we will use the one with grids only for CNNs. 
   - 96 files, ~800 MB each, total 75 GB ~1h30 for generation
   - All targets added (`cl_peptide` (Gibbs), `cl_peptide2` (Marieke), `cl_peptide2_10set`, `cl_allele`, `allele_type`)

Final experiments (compare for shuffled, cl_peptide2, and cl_allele configurations)
- `exp_100k_std_bs64_naivegnn_[...]_0_230531`
   - Run for shuffled, cl_peptide2, and cl_allele 
   - NaiveGNN
   - Standardization only
   - run 46 epochs, ~13h total (1 GPU, 16 workers)
- `exp_100k_std_transf_bs64_naivegnn_[...]_0_230531`
   - Run for shuffled, cl_peptide2, and cl_allele
   - NaiveGNN
   - Transformations before applying standardization, as stated in #130
   - run 46 epochs, ~8h total (1 GPU, 16 workers)
   - Slightly improves compared to std only (see `trained_models/deeprank2/experiments/comparisons/`)
- `exp_100k_std_transf_bs64_naivegnn1_[...]_0_230606`
   - Run for shuffled, cl_peptide2, and cl_allele
   - NaiveGNN1 (added one convolutional layer and one linear layer)
   - Transformations before applying standardization, as stated in #130
   - run 47 epochs, ~6h total (1 GPU, 16 workers)
   - No substantial difference for shuffled and peptide clustered data, but improves ~3% for allele clustered data
- `exp_100k_std_transf_bs64_naivegnn1_wloss_[...]_0_230607`
   - Run for shuffled, cl_peptide2, and cl_allele
   - NaiveGNN1 (added one convolutional layer and one linear layer)
   - Transformations before applying standardization, as stated in #130
   - Weighted loss function
   - run 47 epochs, ~7h total (1 GPU, 16 workers)
   - No substantial difference for shuffled and peptide clustered data, but improves ~1% for allele clustered data
- `exp_100k_std_transf_bs64_naivegnn1_wloss_wdecay_[...]_0_230608`
   - Run for shuffled, cl_peptide2, and cl_allele
   - NaiveGNN1 (added one convolutional layer and one linear layer)
   - Transformations before applying standardization, as stated in #130
   - Weighted loss function
   - Weight decay
   - run 70 epochs, ~12h total (1 GPU, 16 workers)

Below are the AUC plots for all the above mentioned models, for each dataset configuration (shuffled, peptides clustered, alleles clustered):

<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/66b1288e-a77d-48d7-9a25-39ae7dc80938">

<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/4f254e40-8ba0-4519-923f-0713790d5768">

<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/03acc113-fdcb-4f33-a943-0094d538d63a">

Even if the differences are not so significant, the best experiment (especially for the alleles cluster) is `exp_100k_std_transf_bs64_naivegnn1_wloss_[...]_0_230607`.

Then we did `exp_100k_std_transf_bs64_naivegnn1_wloss_[...]_0_[date]` for all the configurations of interest, which are shuffled, `cl_peptide` (Gibbs), `cl_peptide2` (Marieke), `cl_peptide2_10set`, `cl_allele` and we compared the results (threshold maximized on validation set): 

<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/18f41e63-f60b-4f18-867a-8b3a93679b64">
<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/f79dc467-5eb2-4414-af6a-45601b468194">

We also re-trained a model on all data (no validation, no testing) using the parameters of `exp_100k_std_transf_bs64_naivegnn1_wloss_[...]_0_[date]` and the best epoch from the shuffled configuration (i.e., looking at the loss curve the 34th), and these are the curves obtained (exp `exp_100k_std_transf_bs64_naivegnn1_wloss_all_data_0_231006`):
<img width="450" alt="image" src="https://github.com/DeepRank/3D-Vac/assets/55382553/57d04cda-bf5b-4914-b692-6b0285deb4c5">

- [x] Decide which peptides' cluster to use (new one or old one?)
   - None.
- [x] We retrained the GNN 5 times for each data configuration (`exp_100k_std_transf_bs64_naivegnn1_wloss_n_run_X_0_240605`, `exp_100k_std_transf_bs64_naivegnn1_wloss_cl_allele_run_X_1_240612`)
- [ ] Rerun final configurations at their best epoch (it will slightly improve the results) - When Snellius will be back running fast - NOT TO DO ANYMORE
